File: branches/Medusa/Source/Fantasy.Base/AddIn.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Configuration;
using System.Xml;

namespace Fantasy
{
    public class AddIn : IConfigurationSectionHandler
    {
        #region IConfigurationSectionHandler Members

        public object Create(object parent, object configContext, System.Xml.XmlNode section)
        {
            return section;
        }

        #endregion


        public static T[] CreateObjects<T>(string xpath)
        {
            object[] temp = CreateObjects(xpath);
            T[] rs = new T[temp.Length];
            temp.CopyTo(rs, 0);
            return rs;
        }

        public static Type[] GetTypes(string xpath)
        {
           
            XmlElement root = (XmlElement)ConfigurationManager.GetSection("addin");

            xpath += "/@type";
            

            List<Type> rs = new List<Type>();

            foreach(XmlNode node in root.SelectNodes(xpath))
            {
                rs.Add(Type.GetType(node.Value, true));
            }
            return rs.ToArray();
        }

        public static object[] CreateObjects(string xpath)
        {
            Type[] types = GetTypes(xpath);
            object[] rs;
            if (types != null)
            {
                rs = new object[types.Length];
                for (int i = 0; i < rs.Length; i++)
                {
                    rs[i] = Activator.CreateInstance(types[i]);
                }
            }
            else
            {
                rs = new object[0];
            }

            return rs;
        }
    }
}

File: AutoConverted/Fantasy.Base/AddIn.java
package Fantasy;

public class AddIn implements IConfigurationSectionHandler
{
//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#region IConfigurationSectionHandler Members

	public final Object Create(Object parent, Object configContext, System.Xml.XmlNode section)
	{
		return section;
	}

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#endregion


	public static <T> T[] CreateObjects(String xpath)
	{
		Object[] temp = CreateObjects(xpath);
		T[] rs = new T[temp.length];
		temp.CopyTo(rs, 0);
		return rs;
	}

	public static java.lang.Class[] GetTypes(String xpath)
	{

		XmlElement root = (XmlElement)ConfigurationManager.GetSection("addin");

		xpath += "/@type";


		java.util.ArrayList<java.lang.Class> rs = new java.util.ArrayList<java.lang.Class>();

		for(XmlNode node : root.SelectNodes(xpath))
		{
			rs.add(java.lang.Class.forName(node.getValue(), true));
		}
		return rs.toArray(new java.lang.Class[]{});
	}

	public static Object[] CreateObjects(String xpath)
	{
		java.lang.Class[] types = GetTypes(xpath);
		Object[] rs;
		if (types != null)
		{
			rs = new Object[types.length];
			for (int i = 0; i < rs.length; i++)
			{
				rs[i] = Activator.CreateInstance(types[i]);
			}
		}
		else
		{
			rs = new Object[0];
		}

		return rs;
	}
}